### ISSUE TITLE
[Feature]: Add a Hugging Face Accelerate example and guide

### ISSUE THREAD

**Issue by abhinavsriva** [traceopt-ai/traceml#190]
### What would you like TraceML to do?

Add a small, tested example showing how to use TraceML with a Hugging Face Accelerate training loop.

TraceML already supports Hugging Face Trainer through `TraceMLTrainerCallback`, but Accelerate users often write their own PyTorch loop with `Accelerator.prepare(...)` and `accelerator.backward(...)`. We should document that path clearly.

This should be an example/guide first, not a new callback API.

Suggested scope:

- Add `examples/integrations/accelerate_minimal.py`.
- Add `docs/user_guide/integrations/accelerate.md`.
- Link the guide from `docs/user_guide/integrations.md`.
- Link the example from `examples/README.md`.
- Keep `accelerate` optional; do not add it as a core dependency.
- Add a small import-skipped test or smoke test so CI does not require Accelerate by default.

The guide should also explain the launch path. For the first version, the documented path should be:
```
traceml run examples/integrations/accelerate_minimal.py --mode=summary
```
and for single-node multi-GPU:
```
traceml run examples/integrations/accelerate_minimal.py --mode=summary --nproc-per-node=2
```

Out of scope for this first issue:
- Making traceml run wrap accelerate launch.
- Supporting every Accelerate config mode.
- DeepSpeed/FSDP-through-Accelerate tuning.
- A new TraceMLAccelerateCallback API.

Acceptance criteria:
- The example runs on CPU or single GPU.
- The docs explain where traceml.init(...) and traceml.trace_step(...) go.
- The docs explain that TraceML should wrap the prepared model returned by accelerator.prepare(...).
- The docs mention that accelerate launch is not the first supported TraceML path; use traceml run for now.
- The docs call out gradient accumulation briefly, either with a simple note or by leaving it out of the minimal example.
- The final summary is produced successfully.

### Why would this help?

A lot of Hugging Face users use Accelerate when they want more control than `Trainer` gives them. Today TraceML has a clear Trainer path, but not a clear Accelerate path.

This would make TraceML easier to try for custom fine-tuning loops, small distributed experiments, and users who already have an Accelerate script.

**Comment by Animesh452** [comment]
Hi, I'd like to work on this. I'll add examples/integrations/accelerate_minimal.py with a minimal Accelerate training loop, the docs/user_guide/integrations/accelerate.md guide, link both from the existing files, and add an import-skipped smoke test. Let me know if I can go ahead.

**Comment by abhinavsriva** [comment]
@Animesh452 Please go ahead.
